### DISCUSSION TITLE
Add metadata to the job run

### DISCUSSION THREAD

**Discussion by podviaznikov** [dagster-io/dagster#18500] [Q&A]
Is it possible to add metadata to the job after it finished. eg how many DBT models it had etc?

Similar to the code below but for jobs and not assets:

```
context.add_output_metadata(
                metadata={
                    "total_models": total_models,
                },
                output_name=dagster_event.output_name,
            )
```

**Comment by podviaznikov** [comment]
I think this one is related https://github.com/dagster-io/dagster/discussions/18340 and the answer is "no".

But I'm wondering how how to store metadata about each DBT job.

**Reply by yuhan** [reply]
For adding metadata after a run finished, here's an experimental API `report_runless_asset_event`:
```python
from dagster import DagsterInstance, AssetMaterialization

instance = DagsterInstance.get()
instance.report_runless_asset_event(AssetMaterialization(asset_key=..., metadata=...))
```

**Reply by podviaznikov** [reply]
@yuhan but this will save metadata on some asset, right? not on the job run itself. Or basically you are saying I can produce assets dynamically?
